### PR TITLE
Update Embed with SafeArea

### PR DESCRIPTION
**Pull request by codecaffeine**
The adds a `safeArea` parameter to `embed`. When set to true, it will use the safe area layout anchors.

## Food for Thought
Right now I’m defaulting `safeArea` to false so this change doesn’t affect anything currently. I’m very tempted to default it to `true` just to make embed a bit safer without needing any code change. But this would require us to vet our code we’ve currently written to make sure it doesn't mess anything up.

Thoughts?

Also, this change moves the file from #14 so it’s included on `pod install`. It also adds another embed convenience to set a constant inset instead of using UIEdgeInsets

### CODE REVIEW

**Comment by ryang1428** [comment]
Would there ever be a case where you wanted to pin only 1 side to a the safe area? i.e. just the top?

**Comment by codecaffeine** [comment]
@ryang1428 There is a major use for that. I’ve been tempted to write something, but I haven’t come up with and API I particularly like

**Comment by ryang1428** [comment]
how about some sort of struct.. i.e. (rough pseudo code)
```
struct SafeAreaSides {
  static let top: Bool
  static let left: Bool
  static let bottom: Bool
  static let right Bool

  static let allSides = SafeAreaSides(top: true, left: true, bottom: true, right: true) 
}
```

Then in your embed code you base which safe area sides get set looking at that struct?

**Comment by codecaffeine** [comment]
@ryang1428 yeah, I was going around that idea, but a step further:
```swift

struct Edges {
    struct Attach {
        let spacing: CGFloat
        let useSafeArea: Bool
    }

    let leading: Attach?
    let trailing: Attach?
    let top: Attach?
    let bottom: Attach?
}

embed(subView, edges: Edges(leading: Attach(), trailing: Attach(spacing: 20, useSafeArea: true)))
```

**Comment by codecaffeine** [comment]
@ryang1428 all that said, _this_ PR is something I have a need for currently.

**Comment by ryang1428** [comment]
I don't hate that... but it would require some heavy refactor in WL. 

I meant to throw an example call of what I was thinking. This way nothing changes if you don't care about safe areas, and the old signature pretty much stays in tact

`embed(subview: view, insets: .zero, safeAreaSides: SafeAreaSides(top: true, left: false, bottom: false, right: false)`

**Comment by codecaffeine** [comment]
@ryang1428 What do you think about _that_ update?

**Comment by ryang1428** [comment]
ooo I dig that @codecaffeine 👍

File: URBNSwiftyConvenience/Classes/UIView+Additions.swift [hunk 1]
```diff
@@ -8,20 +8,53 @@
 
 import UIKit
 
+public struct SafeAreaEdges: OptionSet {
```
Comment (jgrandelli) [inline URBNSwiftyConvenience/Classes/UIView+Additions.swift]: In the recent past we have decided against `OptionSet`'s for very similar use cases. Why use it here.

Comment (codecaffeine) [reply URBNSwiftyConvenience/Classes/UIView+Additions.swift]: Why did we decide against them?

**Pull request merged**